### PR TITLE
[Hub Generated] Review request for Microsoft.Consumption to add version stable/2019-11-01

### PR DESCRIPTION
**Pull request by sofisa16**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/sofisa16/azure-rest-api-specs/sopadill-consumption-Microsoft.Consumption-2019-11-01...master/).

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “**WaitForARMFeedback**” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://msazure.visualstudio.com/One/_workitems/create/Scenario?templateId=d1c215ec-5d6a-4cb3-a193-95ddd0a1ba96&ownerId=85bb92fb-21fe-41ca-9309-47f2c5f57515). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by sofisa16** [comment]
I am addressing this issues listed here:
https://msazure.visualstudio.com/Swagger%20KPIs/_workitems/edit/8188023

**Comment by j5lim** [comment]
@sofisa16 What are the changes introduced in the new api version? I see only version number changes.

**Comment by sofisa16** [comment]
This is a new version, which has the same information. Nothing changed.

**Comment by jorgecotillo** [comment]
> This is a new version, which has the same information. Nothing changed.

@sofisa16 if nothing changed, what's the motivation to create a new api-version? I saw a comment above saying that this PR intends to correct some Swagger Completeness issues?

**Comment by sofisa16** [comment]
Because we are offering 2019-11-01 version of API to the public without documentation. The customers that are using the version 2019-11-01 are not finding the documentation because it does not exists. We need to offer the documentation to the customers. That is why I created the new swagger for the new api-version.

**Comment by jorgecotillo** [comment]
> Because we are offering 2019-11-01 version of API to the public without documentation. The customers that are using the version 2019-11-01 are not finding the documentation because it does not exists. We need to offer the documentation to the customers. That is why I created the new swagger for the new api-version.

Ok, so your Service supports API 2019-11-01 but there was no swagger available for it, got it. I just checked your manifest and found the version. Signing off from ARM side.

**Comment by sofisa16** [comment]
Here it is for Tags API:

https://armmanifest.visualstudio.com/One/_git/Manifest?path=%2FMICROSOFT.CONSUMPTION%2FPROD%2FMANIFEST.JSON&version=GBmaster&line=1143&lineEnd=1144&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

**Comment by jorgecotillo** [comment]
Based on what I see you have support for some, not all of them, for example budgets:

Your manifest doesn't show 2019-11-01:
https://armmanifest.visualstudio.com/One/_git/Manifest?path=%2FMICROSOFT.CONSUMPTION%2FPROD%2FMANIFEST.JSON&version=GBmaster&line=933&lineEnd=934&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

And you included it in your PR:
https://github.com/Azure/azure-rest-api-specs/blob/afc08a55a22c53548d82f875bb94358f4098d3bf/specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-11-01/consumption.json#L259

Sorry, I might be missing something here.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-11-01/consumption.json [hunk 1]
```diff
@@ -0,0 +1,1278 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2019-11-01",
```
Comment (jorgecotillo) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-11-01/consumption.json:4]: >"2019-11-01" [](start = 15, length = 12)

Any reason why not choosing a more recent year as part of the api version?

**Comment by jorgecotillo** [comment]
>           "default": {

Default should always come as part of a response. See [here](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html)

---
Refers to: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-11-01/consumption.json:468 in 38c457f. [](commit_id = 38c457f4015aff3264cf059f099e0583f6dabb1e, deletion_comment = True)

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json [hunk 2]
```diff
@@ -1829,616 +1829,6 @@
         }
       }
     },
-    "LegacyUsageDetail": {
```
Comment (jorgecotillo) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json:1832]: >"LegacyUsageDetail" [](start = 4, length = 19)

You are updating the contract of an existing api version, you either need to get a breaking change exception or move to a new api.

**Pull request closed without merging**